Feature/create template

mirestrepo opened the pull request.

mirestrepo committed:
feat: add files

mirestrepo committed:
chore: run pre-commit hooks

mirestrepo committed:
ci: remove GCP specifics from action

mirestrepo committed:
fix: echo test

mirestrepo committed:
tests: fixed misslabeled variables

mirestrepo committed:
docs: [skip ci] renamed repo changes

broarr commented on .github/release-drafter.yml:
```diff
@@ -0,0 +1,33 @@
+      - 'bugfix'
+      - 'bug'
+  - title: '🧰 Maintenance'
+    label: 'chore'
```
Is this all commitizen stuff?

mirestrepo replied:
Sort of. It uses the labels of a PR to draft the notes for a release. The label of the PR is given by another action according to the branch name. The notes [here](https://github.com/BrownUniversity/terraform-gcp-project/releases) were made by this release-drafter action. For instance This branch was called `feature/create-template` on PR it was given the label `feature` on merge it will be added to the list of features in the release draft

broarr commented on Gemfile:
```diff
@@ -0,0 +1,4 @@
+source "https://rubygems.org/" do
+    gem "kitchen-terraform", "~> 5.3"
+    gem "inspec-bin", "~> 4.18"
+  end
```
Nit: spacing

mirestrepo replied:
again ;)!

broarr commented on README.md:
```diff
@@ -1,2 +1,108 @@
+## How to use this module
+
+This repository defines a [Terraform module](https://www.terraform.io/docs/modules/usage.html), which you can use in your
+code by adding a `module` configuration and setting its `source` parameter to URL of this repository. See the [examples](/examples) folder for guidance
```
This text should probably include the workflow for copying the template and customizing it, right?

mirestrepo replied:
That's a great point!

broarr commented on test/integration/simple-template/controls/project.rb:
```diff
@@ -0,0 +1,7 @@
+title "Test simple template"
+
+message = attribute("message")
```
Maybe some comments here letting the user know what the attributes are and how they relate to terraform?

broarr commented on test/integration/simple-template/libraries/echo.rb:
```diff
@@ -0,0 +1,22 @@
+
+class Echo < Inspec.resource(1)
```
Maybe some comments here about why and when to create custom Inspec resources?

broarr reviewed (approved):
This is rad! Small cosmetic and documentation changes requested, but nothing super important. I love this template

broestls commented on README.md:
```diff
@@ -1,2 +1,108 @@
+### CI
+This project has three workflows enabled:
+
+1. PR labeler: When openning a PR to defaukt branch, a label is given assigned automatically accourding to the name of your feature branch. The labeler follows the follows rules in [pr-labeler.yml](.github/pr-labeler.yml)
```
Spelling, only a lil thing.

mirestrepo replied:
That's a lot of typos in one paragraph! 😱 I promise I wasn't drinking ;)

broestls commented on versions.tf:
```diff
@@ -0,0 +1,7 @@
+  required_version = ">= 0.12"
+
+  required_providers {
+    google = ">= 3.0, <4.0.0"
```
Isn't `"~> 3.0" == ">=3.0, <4.0.0"`? I think that should cover the version constraint.

broestls commented on versions.tf:
```diff
@@ -0,0 +1,7 @@
+terraform {
+  required_version = ">= 0.12"
```
Terraform 0.13 is probably going to be in a .0 release before the end of the summer. I don't think it would be a bad practice to constrain to `~> 0.12` until we see what the changes look like and if it could be a stumbling block for new users.

broestls reviewed (approved):
I think this looks great! Appreciate the amount of documentation, think it'll be really helpful for users trying to migrate to git for management of their resources.

mirestrepo committed:
chore: improve docs and formatting according to PR review

mirestrepo committed:
chore: change constrain on terraform version, remove google provider, update docs

mirestrepo commented:
@broarr @broestls I think I have addressed all comments. Let me know. And I tried to be more careful about typos... maybe I need a pre-commit hook for that too! 🤣

broarr reviewed (approved):
:shipit: !

mirestrepo commented:
🚀
